src/energyplus/ForwardTranslator/ForwardTranslateCoilWaterHeatingDesuperheater.cpp

```diff
@@ -186,9 +189,32 @@ namespace energyplus {
           idfObject.setString(Coil_WaterHeating_DesuperheaterFields::HeatingSourceObjectType, objectType);
           idfObject.setString(Coil_WaterHeating_DesuperheaterFields::HeatingSourceName, objectName);
         }
+
+        if (ratedHeatReclaimRecoveryEfficiency) {
+          if ((_heatingSource->iddObject().type() == IddObjectType::Refrigeration_Condenser_AirCooled)
+              || (_heatingSource->iddObject().type() == IddObjectType::Refrigeration_Condenser_EvaporativeCooled)
+              || (_heatingSource->iddObject().type() == IddObjectType::Refrigeration_Condenser_WaterCooled)) {
+            if (ratedHeatReclaimRecoveryEfficiency.get() > 0.9) {
+              LOG(Error, modelObject.briefDescription()
+                           << ": Rated Heat Reclaim Recovery Efficiency must be"
+                           << " <= 0.9 when Heating Source Object Type = " << _heatingSource->iddObject().type() << ". It will not be translated.");
+              return boost::none;
```

Comment:
Does this need to happen before we `translateAndMapModelObject` anything?

Reply:
Yeah we probably want to do that check as early as possible.